node/core/av-store/src/lib.rs

```diff
@@ -56,22 +56,37 @@ mod columns {
 
 #[derive(Debug, Error)]
 enum Error {
-	#[error(transparent)]
-	RuntimeAPI(#[from] RuntimeApiError),
 	#[error(transparent)]
 	ChainAPI(#[from] ChainApiError),
 	#[error(transparent)]
 	Erasure(#[from] erasure::Error),
 	#[error(transparent)]
 	Io(#[from] io::Error),
 	#[error(transparent)]
-	Oneshot(#[from] oneshot::Canceled),
+	ChainApiChannelIsClosed(#[from] oneshot::Canceled),
 	#[error(transparent)]
 	Subsystem(#[from] SubsystemError),
 	#[error(transparent)]
 	Time(#[from] SystemTimeError),
 }
 
+/// Class of errors which we should handle more gracefully.
+/// An occurrence of this error should not bring down the subsystem.
+#[derive(Debug, Error)]
```

Comment:
Is this subsystem specific? Or is this generic over all subsystems? I think it is.